### ISSUE TITLE
Odrivetool not working! failure to connect to board.

### ISSUE THREAD

**Issue by Shreeyak** [odriverobotics/ODrive#195]
I just updated to v0.4.1, and the board will not connect to the pc using the odrivetool. The USB can be detected by the system, but the tool hangs while trying to connect to the board. v0.4.0 also does not work. In this state, I find the entire board to be completely unusable.

Please find more details in this thread: https://discourse.odriverobotics.com/t/odrive-doesnt-show-up-in-odrivetool/808/17?u=shreeyak

**Comment by Shreeyak** [comment]
Found some more details:
I ran odrive tool so:
```
 $ sudo -E env "PATH=$PATH" odrivetool -v
ODrive control utility v0.4.1
Waiting for ODrive...
USB discover loop
ConfigurationValue 1
	InterfaceNumber 0,0
		EndpointAddress 130
	InterfaceNumber 1,0
		EndpointAddress 1
		EndpointAddress 129
	InterfaceNumber 2,0
		EndpointAddress 3
		EndpointAddress 131

Please connect your ODrive.
You can also type help() or quit().

Kernel Driver was not attached
EndpointAddress for writing 3
EndpointAddress for reading 131
Connecting to device on USB device bus 1 device 10
In [1]: 

```

I noticed that the odrivetool was resetting the USB into an ACM mode?
```
$ dmesg
...
[17804.495789] usb 1-2: USB disconnect, device number 10
[17808.625756] usb 1-2: new full-speed USB device number 11 using xhci_hcd
[17808.767326] usb 1-2: New USB device found, idVendor=1209, idProduct=0d32
[17808.767327] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[17808.767328] usb 1-2: Product: ODrive 3.4 CDC Interface
[17808.767329] usb 1-2: Manufacturer: ODrive Robotics
[17808.767330] usb 1-2: SerialNumber: 306539843235
[17808.768061] cdc_acm 1-2:1.0: ttyACM0: USB ACM device
[17815.454033] usb 1-2: reset full-speed USB device number 11 using xhci_hcd
[17815.594962] cdc_acm 1-2:1.0: ttyACM0: USB ACM device
```

Here's some other info:
lsusb
```
$ lsusb
Bus 002 Device 004: ID 0451:8140 Texas Instruments, Inc. 
Bus 002 Device 003: ID 0451:8140 Texas Instruments, Inc. 
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 006: ID 1b1c:1b2d Corsair 
Bus 001 Device 004: ID 046d:c332 Logitech, Inc. 
Bus 001 Device 005: ID 0451:8142 Texas Instruments, Inc. TUSB8041 4-Port Hub
Bus 001 Device 003: ID 0451:8142 Texas Instruments, Inc. TUSB8041 4-Port Hub
Bus 001 Device 011: ID 1209:0d32 InterBiometrics 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
lsusb -t
```
$ lsusb -t
/:  Bus 02.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/6p, 5000M
    |__ Port 4: Dev 3, If 0, Class=Hub, Driver=hub/4p, 5000M
        |__ Port 4: Dev 4, If 0, Class=Hub, Driver=hub/2p, 5000M
/:  Bus 01.Port 1: Dev 1, Class=root_hub, Driver=xhci_hcd/12p, 480M
    |__ Port 2: Dev 11, If 0, Class=Communications, Driver=cdc_acm, 12M
    |__ Port 2: Dev 11, If 1, Class=CDC Data, Driver=cdc_acm, 12M
    |__ Port 2: Dev 11, If 2, Class=(Defined at Interface level), Driver=usbfs, 12M
    |__ Port 4: Dev 3, If 0, Class=Hub, Driver=hub/4p, 480M
        |__ Port 4: Dev 5, If 0, Class=Hub, Driver=hub/2p, 480M
    |__ Port 6: Dev 4, If 0, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 6: Dev 4, If 1, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 7: Dev 6, If 0, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 7: Dev 6, If 1, Class=Human Interface Device, Driver=usbhid, 12M
    |__ Port 7: Dev 6, If 2, Class=Human Interface Device, Driver=, 12M
```

lscpi
```
$ lspci -v | grep xhci
	Kernel driver in use: xhci_hcd
```

**Comment by dkraft** [comment]
I hope this helps. I'm having same issue.  uname -a
Linux 8pack 4.15.0-29-generic #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

[ 6834.926322] usb 8-2: new full-speed USB device number 20 using xhci_hcd
[ 6835.335818] usb 8-2: Manufacturer: ODrive Robotics
[ 6835.335819] usb 8-2: SerialNumber: 366533793037
[ 6835.339575] cdc_acm 8-2:1.0: ttyACM0: USB ACM device
[ 6856.742403] usb 8-2: reset full-speed USB device number 20 using xhci_hcd
[ 6857.028139] cdc_acm 8-2:1.0: ttyACM0: USB ACM device
dkraft@8pack:~$ odrivetool -p usb:8:20
ODrive control utility v0.4.1
/usr/local/lib/python3.6/dist-packages/IPython/paths.py:61: UserWarning: IPython dir '/home/dkraft/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Please connect your ODrive.
You can also type help() or quit().

In [1]: quit

dkraft@8pack:~$ odrivetool -p usb:008:020
ODrive control utility v0.4.1
/usr/local/lib/python3.6/dist-packages/IPython/paths.py:61: UserWarning: IPython dir '/home/dkraft/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Please connect your ODrive.
You can also type help() or quit().

In [1]: quit

corrupted size vs. prev_size
^[[AAborted (core dumped)
134 dkraft@8pack:~$ odrivetool -p serial:/dev/ttyACM0
ODrive control utility v0.4.1
/usr/local/lib/python3.6/dist-packages/IPython/paths.py:61: UserWarning: IPython dir '/home/dkraft/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Please connect your ODrive.
You can also type help() or quit().

In [1]: quit

dkraft@8pack:~$ odrivetool -p serial:/dev/ttyACM0
ODrive control utility v0.4.1
/usr/local/lib/python3.6/dist-packages/IPython/paths.py:61: UserWarning: IPython dir '/home/dkraft/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Please connect your ODrive.
You can also type help() or quit().

In [1]: quit

dkraft@8pack:~$ odrivetool -s 366533793037
ODrive control utility v0.4.1
/usr/local/lib/python3.6/dist-packages/IPython/paths.py:61: UserWarning: IPython dir '/home/dkraft/.ipython' is not a writable location, using a temp directory.
  " using a temp directory.".format(ipdir))
Please connect your ODrive.
You can also type help() or quit().

In [1]: quit

**Comment by madcowswe** [comment]
Should be fixed in firmware version 0.4.4

**Comment by Shreeyak** [comment]
Awesome! Thank you! I can now start with ROS development for the ODrive!
